### PR TITLE
Numbanize state space

### PR DESCRIPTION
**Pull request by boryana-ilieva**
With this PR we want to speed up soepy's state space creation.

In order to ensure that the package still functions as intended, changes are to be made in as small and granular steps as possible.

- [x] Numba over pyth_create_state_space and recreate old state space objects through a conversion function.
- [x] Rewrite all remaining functions as to depend on the new state space objects.

### CODE REVIEW

**Comment by boryana-ilieva** [comment]
I am running pr testing on this first step and proceeding with second on Monday.

**Comment by boryana-ilieva** [comment]
Running PR testing on this step.

**Comment by boryana-ilieva** [comment]
@peisenha, I am rerunning PR testing on the whole thing one last time tonight, but I would appreciate a first review already.

File: soepy/python/shared/shared_auxiliary.py [hunk 1]
```diff
@@ -23,61 +23,105 @@ def draw_disturbances(seed, shocks_cov, num_periods, num_draws):
...
 
     # Ensure that the benefits are recorded as non-labor income in data frame
-    period_wages[0] = model_params.benefits
+    period_wages[:, :, 0] = model_params.benefits
 
     # Return function output
-    return (
-        period_wages
-    )  # This is a vector, difference between choices comes from disturbance term.
+    # Dimension (num_states x num_draws x num_choices)
```
Comment (peisenha) [inline soepy/python/shared/shared_auxiliary.py]: Please move these kinds of comments to the docstrings.

Comment (boryana-ilieva) [reply soepy/python/shared/shared_auxiliary.py]: This is already included in the docstring of the wrapper function calculate_utilities, so I will just delete it here, okay?

**Comment by peisenha** [comment]
Sure

**Pull request merged**